### DISCUSSION TITLE
Option for running new GWE model in flopy parallel processing (MF6Splitter)?

### DISCUSSION THREAD

**Discussion by gjkDarcy** [modflowpy/flopy#2269] [Q&A]
I'm currently trying to run a GWF-GWE model (GWF and GWE are separate simulations that are connected) using parallel processing via MF6Splitter, and I'm wondering if it can be used on the GWE model or if that is still in the process of being developed. I don't see a "gwe" option in the source code (only gwf and gwt), and I'm getting the error: "AttributeError: 'ModflowGwenam' object has no attribute 'newtonoptions'".

**Comment by emorway-usgs** [comment]
Hello @gjkDarcy, I'm not entirely sure if [this pull request](https://github.com/MODFLOW-USGS/modflowapi/pull/49) addresses your particular issue, or if some attention needs to be given to the MF6Splitter class, but it is a start since support for GWE was missing from the modflowapi class.  Perhaps if you could provide a bit more information about where exactly in your workflow things go awry, that would help us better identify what we need to fix on our end.  If you have the ability to generate a small reproducible example, that would of course be welcome!

**Comment by gjkDarcy** [comment]
Thank you for looking into this @emorway-usgs! Sorry for not providing additional details in my initial question. I am using flopy 3.8.0 with modflow6 6.5, and I created a gwf-gwe simulation that is testing thermal injection into an multi-layer aquifer via a well injection pump [similar to the interacting borehole heat exchangers example found on the modflow6-examples page](https://github.com/MODFLOW-USGS/modflow6-examples/blob/master/scripts/ex-gwe-geotherm.py). I tried implementing the Mf6Splitter class using the following:


```
# Settings from environment variables
write = get_env("WRITE", run_bool)
run = get_env("RUN", run_bool)
plot = get_env("PLOT", run_bool)


#Build models
sim_mf6gwf, sim_mf6gwe = build_models(sim_name,
                                        workspace,
                                        grid_params=grid_params,
                                        bore_params=bore_params,
                                        model_variable_params=model_variable_params)

#split simulations into smaller portions to be run concurrently (GWF)
gwf_mfsplit = Mf6Splitter(sim_mf6gwf)
gwf_split_array = gwf_mfsplit.optimize_splitting_mask(nparts = 2)
split_gwf_sim = gwf_mfsplit.split_model(gwf_split_array)

#split simulations into smaller portions to be run concurrently (GWE)
gwe_mfsplit = Mf6Splitter(sim_mf6gwe)
gwe_split_array = gwe_mfsplit.optimize_splitting_mask(nparts = 2)
split_gwe_sim = gwe_mfsplit.split_model(gwe_split_array)


# write models
if write:
    split_gwf_sim.write_simulation(silent=silent)
    split_gwe_sim.write_simulation(silent=silent)


# run models
if run:
    run_model(split_gwf_sim)
    run_model(split_gwe_sim)


#reassemble the models (GWF)
gwf_model_names = list(split_gwf_sim.model_names) #get list of split models
head_dict = {}
for modelname in gwf_model_names:
    model_num = int(modelname.split("_")[-1]) # get num ID of model
    head = split_gwf_sim.get_model(modelname).output.head().get_alldata()[-1] #get head data for gwf model
    head_dict[model_num] = head

ra_heads = gwf_mfsplit.reconstruct_array(head_dict) #recombine all head data in the head_dict

#reassemble the models (GWE)
gwe_model_names = list(split_gwe_sim.model_names) # get list of split models
temp_dict = {}
for modelname in gwe_model_names:
    model_num = int(modelname.split("_")[-1]) #get num ID of model
    temp = split_gwe_sim.get_model(modelname).output.temp().get_alldata()[-1] #get temp data for gwe model
    temp_dict[model_num] = temp

ra_temps = gwe_mfsplit.reconstruct_array(temp_dict) #recombine all temp data in the temp_dict
```

The combined gwf-gwe model I created work well when the above code for parallel processing is not implemented, but when I implement it, I get the error: AttributeError: 'ModflowGwenam' object has no attribute 'newtonoptions', which seems like it could stem from the model_splitter.py script not being set up for the gwe model yet? 

The error points to the _create_exchanges(self) function in the Mf6Splitter class in model_splitter.py. I don't think I'm going to be able to put together a simple working example quickly unfortunately. I will try out the additions in the pull request you've linked - although it seems like they are to the modflowapi, and I'm uncertain how to integrate those with the flopy code (sorry - still pretty new to modflow and flopy).

**Comment by emorway-usgs** [comment]
The additional information is very helpful @gjkDarcy.  The fixes (still underway) on the modflowapi PR are not going to solve this particular issue.  I think some additional work on the Splitter will be necessary to get you on your way.  Any chance you could provide the contents of the `build_models()` function, or perhaps attach the full script you're using and we could just use that to debug on our end?  Understandable if not, we could likely find an example to work up on our end.  If helpful, you're welcome share it directly if you prefer not to post it to a public space like this one (emorway at usgs dot gov).

**Reply by gjkDarcy** [reply]
I sent the full code to you privately @emorway-usgs, thank you!
